wallet-new/src/Cardano/Wallet/Server/Plugins.hs

```diff
@@ -163,3 +160,35 @@ updateNotifier :: Plugin Kernel.WalletMode
 updateNotifier = [
     \_diffusion -> logError "Not Implemented: updateNotifier [CBR-374]"
     ]
+
+-- | A @Middleware@ to throttle requests.
+throttleMiddleware :: Maybe ThrottleSettings -> Middleware
+throttleMiddleware Nothing app = app
+throttleMiddleware (Just ts) app = \req respond -> do
+    throttler <- Throttle.initThrottler
+    Throttle.throttle throttleSettings throttler app req respond
+  where
+    throttleSettings = Throttle.defaultThrottleSettings
+        { Throttle.onThrottled = \microsTilRetry ->
+            let
+                err = V1.RequestThrottled microsTilRetry
+            in
+                responseLBS (V1.toHttpErrorStatus err) [applicationJson] (encode err)
+        , Throttle.throttleRate = fromIntegral $ tsRate ts
+        , Throttle.throttlePeriod = fromIntegral $ tsPeriod ts
+        , Throttle.throttleBurst = fromIntegral $ tsBurst ts
+        }
+
+-- | A @Middleware@ to enable CORS.
+-- When running in debug mode, chances are we want to at least allow CORS to test the API
+-- with a Swagger editor, locally.
+corsMiddleware :: RunMode -> Middleware
+corsMiddleware wrm =
+    if isDebugMode wrm
+        then cors (const $ Just policy)
+        else identity
+  where
+    policy = simpleCorsResourcePolicy
+        { corsRequestHeaders = ["Content-Type"]
+        , corsMethods = "PUT" : simpleMethods
+        }
```

Comment:
Note that we don't need this CORS middleware anymore. It seems that it was used only because we had those executable snippets in the documentation in the past (so client code was making XHR requests to the API). That's not the case anymore so let's just remove this.

Reply:
is it safe to remove it from legacy wallet - or we should strive not to modify the old wallet?

Reply:
It is safe too, but I'd rather not do any work in the old wallet anymore :)